di-ipv-cimit-stub/lambdas/get-contra-indicator-credential/src/main/java/uk/gov/di/ipv/core/getcontraindicatorcredential/GetContraIndicatorCredentialHandler.java

```diff
@@ -0,0 +1,116 @@
+
+        SignedJWT signedJWT;
+        try {
+            signedJWT = generateJWT(getValidClaimsSetValues(), event.getUserId());
+        } catch (Exception ex) {
+            LOGGER.error(
+                    new StringMapMessage()
+                            .with(
+                                    "errorDescription",
+                                    "Failed at stub during creation of signedJwt. Error message:"
+                                            + ex.getMessage()));
+            return "Failure";
+        }
+        return signedJWT.serialize();
+    }
+
+    private SignedJWT generateJWT(Map<String, Object> claimsSetValues, String userId)
+            throws NoSuchAlgorithmException, InvalidKeySpecException, JOSEException {
+        ECDSASigner signer = new ECDSASigner(getPrivateKey());
+
+        SignedJWT signedJWT =
+                new SignedJWT(
+                        new JWSHeader.Builder(JWSAlgorithm.ES256).type(JOSEObjectType.JWT).build(),
+                        generateClaimsSet(claimsSetValues, userId));
+        signedJWT.sign(signer);
+
+        return signedJWT;
+    }
+
+    private ECPrivateKey getPrivateKey() throws InvalidKeySpecException, NoSuchAlgorithmException {
+        var EC_PRIVATE_KEY = GetCICredentialConfig.getCimitSigningKey();
+        return (ECPrivateKey)
+                KeyFactory.getInstance("EC")
+                        .generatePrivate(
+                                new PKCS8EncodedKeySpec(
+                                        Base64.getDecoder().decode(EC_PRIVATE_KEY)));
+    }
+
+    private JWTClaimsSet generateClaimsSet(Map<String, Object> claimsSetValues, String userId) {
+        return new JWTClaimsSet.Builder()
+                .claim(JWTClaimNames.SUBJECT, userId)
+                .claim(
+                        JWTClaimNames.ISSUER,
+                        claimsSetValues.get(GetCICredentialConfig.getCimitComponentId()))
+                .claim(JWTClaimNames.NOT_BEFORE, claimsSetValues.get(JWTClaimNames.NOT_BEFORE))
+                .claim(
+                        JWTClaimNames.EXPIRATION_TIME,
+                        claimsSetValues.get(JWTClaimNames.EXPIRATION_TIME))
+                .claim(VC, claimsSetValues.get(VC))
+                .build();
+    }
+
+    private Map<String, Object> getValidClaimsSetValues() {
+        return Map.of(
+                JWTClaimNames.SUBJECT,
+                "sub",
+                JWTClaimNames.ISSUED_AT,
+                OffsetDateTime.now().toEpochSecond(),
+                JWTClaimNames.NOT_BEFORE,
+                OffsetDateTime.now().toEpochSecond(),
+                JWTClaimNames.EXPIRATION_TIME,
+                OffsetDateTime.now().plusSeconds(15 * 60).toEpochSecond(),
+                VC,
+                generateVC());
+    }
+
+    private Map<String, Object> generateVC() {
+        Map<String, Object> vc = new LinkedHashMap<>();
```

Comment:
Should probably be using `com.nimbusds.jwt.JWTClaimsSet` which has a builder - see core-back for examples